### PR TITLE
[FIX] PEP8

### PR DESCRIPTION
**Pull request by pedrobaeza**

### CODE REVIEW

**Comment by eLBati** [comment]
Just 3 unused import

```
./hr_attendance_analysis/hr_attendance.py:26:1: F401 'time' imported but unused
./hr_attendance_analysis/hr_contract.py:31:1: F401 'fields' imported but unused
./hr_attendance_analysis/report/calendar_report.py:24:1: F401 'osv' imported but unused
```

**Comment by pedrobaeza** [comment]
Fixed!

**Comment by eLBati** [comment]
:+1:

File: hr_attendance_analysis/hr_attendance.py [hunk 1]
```diff
@@ -69,36 +70,36 @@ def float_to_datetime(self, float_val):
...
     def time_sum(self, float_first_time, float_second_time):
         str_first_time = self.float_time_convert(float_first_time)
-        first_timedelta = timedelta(0, int(str_first_time.split(':')[0]) * 60.0*60.0 +
+        first_timedelta = timedelta(
+            0, int(str_first_time.split(':')[0]) * 60.0 * 60.0 +
             int(str_first_time.split(':')[1]) * 60.0)
         str_second_time = self.float_time_convert(float_second_time)
-        second_timedelta = timedelta(0, int(str_second_time.split(':')[0]) * 60.0*60.0 +
+        second_timedelta = timedelta(
+            0, int(str_second_time.split(':')[0]) * 60.0 * 60.0 +
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: > `60.0 * 60.0`

`3600.0`

File: hr_attendance_analysis/hr_attendance.py [hunk 2]
```diff
@@ -156,64 +162,69 @@ def get_reference_calendar(self, cr, uid, employee_id, date=None, context=None):
...
-        attendance_pool = self.pool.get('resource.calendar.attendance')
-        precision = self.pool.get('res.users').browse(cr, uid, uid, context=context).company_id.working_time_precision
+        attendance_pool = self.pool['resource.calendar.attendance']
+        precision = self.pool['res.users'].browse(
+            cr, uid, uid, context=context).company_id.working_time_precision
         # 2012.10.16 LF FIX : Get timezone from context
-        active_tz = pytz.timezone(context.get("tz","UTC") if context else "UTC")
+        active_tz = pytz.timezone(
+            context.get("tz", "UTC") if context else "UTC")
         str_now = datetime.strftime(datetime.now(), '%Y-%m-%d %H:%M:%S')
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: `datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)`

File: hr_attendance_analysis/hr_attendance.py [hunk 3]
```diff
@@ -156,64 +162,69 @@ def get_reference_calendar(self, cr, uid, employee_id, date=None, context=None):
...
             duration = 0.0
-            outside_calendar_duration = 0.0
-            inside_calendar_duration = 0.0
             attendance = self.browse(cr, uid, attendance_id, context=context)
             res[attendance.id] = {}
             # 2012.10.16 LF FIX : Attendance in context timezone
             attendance_start = datetime.strptime(
-                attendance.name, '%Y-%m-%d %H:%M:%S'
-                ).replace(tzinfo=pytz.utc).astimezone(active_tz)
+                attendance.name, '%Y-%m-%d %H:%M:%S').replace(
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: `DEFAULT_SERVER_DATETIME_FORMAT`

File: hr_attendance_analysis/hr_attendance.py [hunk 4]
```diff
@@ -156,64 +162,69 @@ def get_reference_calendar(self, cr, uid, employee_id, date=None, context=None):
...
+                        raise orm.except_orm(
+                            _('Error'),
+                            _('Incongruent data: sign-in %s is followed by '
+                              'another sign-in') % attendance_start)
                     next_attendance_date = next_attendance.name
                 # 2012.10.16 LF FIX : Attendance in context timezone
                 attendance_stop = datetime.strptime(
-                    next_attendance_date, '%Y-%m-%d %H:%M:%S'
-                    ).replace(tzinfo=pytz.utc).astimezone(active_tz)
+                    next_attendance_date, '%Y-%m-%d %H:%M:%S').replace(
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: DEFAULT_SERVER_DATETIME_FORMAT

File: hr_attendance_analysis/hr_attendance.py [hunk 5]
```diff
@@ -222,42 +233,43 @@ def _get_attendance_duration(self, cr, uid, ids, field_name, arg, context=None):
...
+                        float_attendance_rounding = float(
+                            reference_calendar.attendance_rounding)
                         rounded_start_hour = self._ceil_rounding(
                             float_attendance_rounding, attendance_start)
                         rounded_stop_hour = self._floor_rounding(
                             float_attendance_rounding, attendance_stop)
-                            
-                        if abs(1- rounded_start_hour) < 0.01: # if shift == 1 hour
-                            attendance_start = datetime(attendance_start.year, attendance_start.month,
+                        # if shift == 1 hour
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: I don't understand the comment.
Would be better to put `# if shift is approximately one hour`

File: hr_attendance_analysis/hr_attendance.py [hunk 6]
```diff
@@ -266,90 +278,115 @@ def _get_attendance_duration(self, cr, uid, ids, field_name, arg, context=None):
...
-                            ('hour_from','<=',centered_attendance_hour),
-                            ], context=context)
+                            centered_attendance.hour +
+                            centered_attendance.minute / 60.0
+                            + centered_attendance.second / 3600.0
+                        )
+                        # check if centered_attendance is within a working
+                        # schedule 2012.10.16 LF FIX : weekday must be single
+                        # character not int
+                        weekday_char = str(
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: `ustr`

File: hr_attendance_analysis/hr_attendance.py [hunk 7]
```diff
@@ -266,90 +278,115 @@ def _get_attendance_duration(self, cr, uid, ids, field_name, arg, context=None):
...
+                             ('hour_to', '>=', centered_attendance_hour),
+                             ('hour_from', '<=', centered_attendance_hour)],
+                            context=context)
                         if len(matched_schedule_ids) > 1:
-                            raise orm.except_orm(_('Error'),
-                                _('Wrongly configured working schedule with id %s') % str(calendar_id))
+                            raise orm.except_orm(
+                                _('Error'),
+                                _('Wrongly configured working schedule with '
+                                  'id %s') % str(calendar_id))
```
Comment (bwrsandman) [inline hr_attendance_analysis/hr_attendance.py]: ustr

File: hr_attendance_analysis/report/calendar_report.py [hunk 8]
```diff
@@ -36,14 +36,13 @@ def _get_day_of_week(self, day):
             4: _('Friday'),
             5: _('Saturday'),
             6: _('Sunday'),
-            }
-        dayofweek=''
-        weekday = datetime.strptime(day,'%Y-%m-%d').weekday()
+        }
+        weekday = datetime.strptime(day, '%Y-%m-%d').weekday()
```
Comment (bwrsandman) [inline hr_attendance_analysis/report/calendar_report.py]: `DEFAULT_SERVER_DATE_FORMAT`

File: hr_attendance_analysis/wizard/print_calendar_report.py [hunk 9]
```diff
@@ -108,81 +113,86 @@ def print_calendar(self, cr, uid, ids, context=None):
...
-                                and (
-                                not calendar_attendance.date_from or 
-                                datetime.strptime(calendar_attendance.date_from,'%Y-%m-%d')
-                                <= current_date
-                                )):
-                                calendar_attendance_duration = attendance_pool.time_difference(
-                                    calendar_attendance.hour_from, calendar_attendance.hour_to)
+                        for calendar_attendance in \
+                                reference_calendar.attendance_ids:
+                            if ((not calendar_attendance.dayofweek
```
Comment (bwrsandman) [inline hr_attendance_analysis/wizard/print_calendar_report.py]: This is far too indented, recommend factoring into dedicated functions or combine ifs with and

**Comment by eLBati** [comment]
Some improvements at https://github.com/pedrobaeza/hr-timesheet/pull/1

**Comment by pedrobaeza** [comment]
Fixed your comments, @bwrsandman.

Thanks.

**Comment by yvaucher** [comment]
:+1:

**Pull request merged**